modules/library/referencing/src/main/java/org/geotools/referencing/CRS.java

```diff
@@ -266,6 +266,25 @@ private static boolean isForcedLonLat() {
         return FORCED_LON_LAT.get();
     }
 
+    /**
+     * Determines if two coordinate reference systems are compatible with each other, i.e. can be
+     * converted from one to another in a meaningful way.
+     *
+     * <p>They must have the same dimension (see OGC 09-025r2).
```

Comment:
Nope! I believe there are some code paths allowing it to go from a 2D+1 CRS down to its equivalent 2D (basically just stripping the Z) when the 2D part is common to the two. And while others are missing today, they could be added tomorrow or be present in specific usage scenarios (transformations are pluggable).
It's probably more robust to go and search for a math transform and catch an exception if not found... since catching an exception can be expensive, would likely be a good idea to first check if the two CRSs are not already ``equalsIgnoreMetadata``.

Reply:
Okay. The specs I referred to elsewhere was not very clear about compatibility, and mentioned only different dimensions as a clear example. I indeed wanted to avoid it being expensive. Especially because this method is called from three different places (reprojecting the type as well as the feature, and first even to check if a reprojectfeaturereader should be created at all). It would be ideal if we only attempted to create the mathtransform once and caught the exception once. I will look into it.

It should in any case then be made more clear then that the order of arguments in this method matters.